libraries/clocker.rb

```diff
@@ -0,0 +1,116 @@
+class Clocker
+  class ZKs
+    def initialize
+      # hash of zookeeper locks we use one connection per lock id
+      @@zkls = {}
+      return
+    end
+
+    # Create or retrieve a locker as appriate
+    def locker(lock, zookeeper = nil)
+      require 'zk'
+      if @@zkls.include?(lock)
+        Chef::Log.debug("Returning locker for #{lock}")
+        @@zkls[lock]
+      elsif zookeeper
+        Chef::Log.debug("Creating zookeeper connection for #{lock}")
+        zkc = ZK.new(zookeeper)
+        @@zkls[lock] = ZK.new(zookeeper)
+        @@zkls[lock] = zkc.locker(lock)
+      else
+        # Lock neither found nor created
+        Chef::Log.debug("No lock found for #{lock}")
+        false
+      end
+    end
+
+    def disco(lock)
+      Chef::Log.debug("Disconecting #{lock}'s zookeeper connection")
+      if @@zkls.include?(lock)
+         @@zkls[lock].close
```

Comment:
I think it's sane to leave the closed locker in the hash, if for some reason somebody use the [Clocker#held?](https://github.com/jameslegg/chef-locker/blob/use-zk-lock-recipe/libraries/clocker.rb#L44) method later in the chef run it will correctly return false after checking [ZK::ExclusiveLocker#assert!](http://www.rubydoc.info/gems/zk/ZK/Locker/ExclusiveLocker#assert%21-instance_method) when asked. 

I've got a basic test for that behaviour in the test cookbook in jenkins.rb recipe 
https://github.com/jameslegg/chef-locker/blob/use-zk-lock-recipe/test/cookbooks/cl_test/recipes/jenkins.rb#L28

And a serverspec test to verify:

https://github.com/jameslegg/chef-locker/blob/use-zk-lock-recipe/test/integration/jenkins/serverspec/default_spec.rb#L15